pkg/fileutils/read.go

```diff
@@ -94,3 +99,12 @@ func downloadFile(url string, timeout time.Duration) ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+func onWindowsSchemeIsDriveLetter(u *url.URL) bool {
+	// On Windows the URL potentially ends up with the drive letter as the scheme so compare with the URL as a string
+	if runtime.GOOS == "windows" && len(u.Scheme) == 1 && strings.HasPrefix(u.String(), u.Scheme + ":") {
```

Comment:
Doesn't `len(u.Scheme) == 1` _imply_ that `strings.HasPrefix(u.String(), u.Scheme + ":")`? If not, we need some testcases which show why this extra conjunct is needed.

Actually, more tests in this area, to cover the Windows novelties are probably advisable anyway, like Florian said.

Reply:
Even, isn't `strings.HasPrefix(u.String(), u.Scheme + ":")` ALWAYS true? This is just testing that a URL starts with its scheme followed by a colon. I don't think there is a 100% sure way to test this (there could always be a fancy scheme named "d"), but testing Windows + 1 letter scheme seems to be the best we can do for now

Reply:
Looking at the whole chain the manifest file reference goes through, it seems we massage the path and strip out the scheme from the file reference, then we parse it into a URL again in this function, so now something like `C:/foo/bar.yaml` gets parsed as a scheme of `c` and a path of `/foo/bar.yaml` and a file like `C:\foo\bar.yaml` gets a scheme of `c` and and empty path. Lots of weird issues but this fix fixes some. And, yeah, we probably don't need the prefix test as long as the scheme is 1 character.